### PR TITLE
change(deprecated_functions): replace cx_hash with cx_hash_no_throw a…

### PR DESCRIPTION
**Pull request by ndedouhet-ledger**
…nd cx_aes with cx_aes_no_throw

### CODE REVIEW

File: src/backend/bolos-vaultapp/hash-sha256.c [hunk 1]
```diff
@@ -50,14 +50,14 @@ static void noise_sha256_update(NoiseHashState *state, const uint8_t *data, size
 {
     NoiseSHA256State *st = (NoiseSHA256State *)state;
 
-    cx_hash(&(st->sha256.header), 0, data, len, NULL, 0);
+    cx_hash_no_throw(&(st->sha256.header), 0, data, len, NULL, 0);
```
Comment (pandouard) [inline src/backend/bolos-vaultapp/hash-sha256.c]: Shouldn't the return status be checked?

Comment (ndedouhet-ledger) [reply src/backend/bolos-vaultapp/hash-sha256.c]: Nice catch!

**Review by jbeaussart-ledger — CHANGES_REQUESTED** [review]
Some functions have now a return value to check

File: src/backend/bolos-vaultapp/hash-sha256.c (hunk 1 above)
Comment (ndedouhet-ledger) [reply src/backend/bolos-vaultapp/hash-sha256.c]: Here we will ignore return value to not break the noise design pattern

**Review by jbeaussart-ledger — APPROVED** [review]
LGTM

**Pull request merged**